kokain-processor/src/main/java/com/schwarz/kokain/processor/model/EFactoryModel.kt

```diff
@@ -1,38 +1,32 @@
 package com.schwarz.kokain.processor.model
 
 import com.schwarz.kokain.api.EFactory
-import com.squareup.kotlinpoet.ClassName
+import com.schwarz.kokain.kokaingeneratorlib.model.IEFactoryModel
 import com.squareup.kotlinpoet.TypeName
+import com.squareup.kotlinpoet.asTypeName
 import com.sun.tools.javac.code.Symbol
 import javax.lang.model.element.Element
 import javax.lang.model.type.MirroredTypesException
 import javax.lang.model.type.TypeMirror
 import javax.lang.model.util.Elements
 
-class EFactoryModel(sourceElement: Element, util: Elements) {
+class EFactoryModel(sourceElement: Element, util: Elements) : IEFactoryModel {
 
-    var additionalFactories: Array<TypeMirror>
+    var additionalFactoriesTypeMirror: Array<TypeMirror> = sourceElement.getAnnotation(EFactory::class.java).let {
```

Comment:
would make sense to use lazy-initialization here (and on line 25) so we only need to execute this when it is used?

Reply:
in general yes but I think the kokain-processor will become obsolete anyway when we verified that the ksp way is working fine. Or at least that was my plan since I see no advantages for keeping them both alive do you have a opinion on that?

Reply:
Valid point 👍 (except for the exception-thing in the other comment)
And yes, doing the same thing twice probably makes little sense :-)